gadgets/snapshot_process/test/snapshot_process_test.go

```diff
@@ -0,0 +1,116 @@
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type snapshotProcessEntry struct {
+	eventtypes.CommonData
+
+	MountNsID uint64 `json:"mntns_id"`
+	Pid       uint32 `json:"pid"`
+	Tid       uint32 `json:"tid"`
+	Ppid      uint32 `json:"ppid"`
+	Uid       uint32 `json:"uid"`
+	Gid       uint32 `json:"gid"`
+	Comm      string `json:"comm"`
+}
+
+func TestSnapshotProcess(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-snapshot-process"
+	containerImage := "docker.io/library/busybox:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{containers.WithContainerImage(containerImage)}
+
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-snapshot-process")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+
+	testContainer := containerFactory.NewContainer(
+		containerName,
+		"setuidgid 1000:1111 nc -l -p 9090",
+		containerOpts...,
+	)
+
+	testContainer.Start(t)
+	t.Cleanup(func() {
+		testContainer.Stop(t)
+	})
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{utils.WithContainerImageName(containerImage), utils.WithContainerID(testContainer.ID())}
+
+	// TODO: timeout shouldn't be required
+	switch utils.CurrentTestComponent {
+	case utils.IgLocalTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-r=%s", utils.Runtime), "--timeout=1"))
+	case utils.KubectlGadgetTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-n=%s", ns), "--timeout=1"))
+		testingOpts = append(testingOpts, igtesting.WithCbBeforeCleanup(utils.PrintLogsFn(ns)))
+		commonDataOpts = append(commonDataOpts, utils.WithK8sNamespace(ns))
+	}
+
+	runnerOpts = append(runnerOpts, igrunner.WithValidateOutput(
+		func(t *testing.T, output string) {
+			expectedEntry := &snapshotProcessEntry{
+				CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+				Comm:       "nc",
+				Uid:        1000,
+				Gid:        1111,
+
+				// Check the existence of the following fields
+				MountNsID: utils.NormalizedInt,
+				Pid:       utils.NormalizedInt,
+				Tid:       utils.NormalizedInt,
+				Ppid:      utils.NormalizedInt,
+			}
+
+			normalize := func(e *snapshotProcessEntry) {
+				utils.NormalizeCommonData(&e.CommonData)
+				utils.NormalizeInt(&e.MountNsID)
+				utils.NormalizeInt(&e.Pid)
+				utils.NormalizeInt(&e.Tid)
+				utils.NormalizeInt(&e.Ppid)
+			}
+
+			match.MatchEntries(t, match.PerLineJSONArrayMode, output, normalize, expectedEntry)
```

Comment:
Mmmm using `PerLineJSONArrayMode` for snapshotters prevent us from verifying they are printing output in one single array. It means that if a snapshotter prints outcomes from each node separately, one per line, then this test will pass, and we don't want that. 

So, for now that we're not merging outcome from all nodes into one array it's fine, but probably we should use `JSONArrayMode` for snapshotters. WDYT? If you agree, please add a TODO here to change once the combiner operator is merged.

Reply:
You're right, this was working because we were running the integration tests in a cluster with a single node. I added a commit to this PR to use multiple nodes.